internal/setup/setup.go

```diff
@@ -86,13 +86,15 @@ func Setup(ctx context.Context, config DBConfigProvider) (*serverenv.ServerEnv,
 		}
 		opts = append(opts, serverenv.WithKeyManager(km))
 	}
-	// TODO(mikehelmick): Make this extensible to other providers.
+
 	if _, ok := config.(BlobStorageConfigProvider); ok {
-		storage, err := storage.NewGoogleCloudStorage(ctx)
+		provider := config.(BlobStorageConfigProvider)
```

Comment:
In Go we're usually suspicious of providers and factories. 
If you're trying to create a particular kind of blob store based on a string (coming from the environment, or wherever), then all you need is
```
switch blobStoreKind {
case "gcs":
   return gcsblobstore.New(...)
case "file":
   return fileblobstore.New(...)
}
```

Reply:
Fixed.
Even if I inline the factory code it still remains factory code. =)

Reply:
I think of everything in an API as having a weight. Struct fields have less weight than global functions, which have less weight than concrete structs, which have less weight than interfaces. I strive to minimize weight. A single top-level "factory" function is lighter than a Factory interface implemented by multiple structs each with their own methods. (Not that you had something that complicated—I honestly didn't look that carefully—but that is what tends to happen.)

Anyway, thanks for simplifying.